src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsauditapi/services/AuditS3Client.kt

```diff
@@ -0,0 +1,47 @@
+package uk.gov.justice.digital.hmpps.hmppsauditapi.services
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+import software.amazon.awssdk.core.sync.RequestBody
+import software.amazon.awssdk.services.s3.S3Client
+import software.amazon.awssdk.services.s3.model.PutObjectRequest
+import uk.gov.justice.digital.hmpps.hmppsauditapi.listeners.HMPPSAuditListener
+import java.nio.charset.StandardCharsets
+import java.security.MessageDigest
+import java.time.ZoneId
+
+@Service
+class AuditS3Client(
+  private val s3Client: S3Client,
+  private val objectMapper: ObjectMapper,
+  @Value("\${aws.s3.auditBucketName}") private val bucketName: String,
+) {
+
+  fun save(auditEvent: HMPPSAuditListener.AuditEvent) {
+    val auditEventJsonString = objectMapper.writeValueAsString(auditEvent)
+    val fileName = generateFilename(auditEvent, auditEventJsonString)
+
+    val putObjectRequest = PutObjectRequest.builder()
+      .bucket(bucketName)
+      .key(fileName)
+      .build()
+
+    val jsonBytes = auditEventJsonString.toByteArray(StandardCharsets.UTF_8)
+
+    s3Client.putObject(putObjectRequest, RequestBody.fromBytes(jsonBytes))
+  }
+
+  private fun generateFilename(auditEvent: HMPPSAuditListener.AuditEvent, auditEventJsonString: String): String {
+    val whenDateTime = auditEvent.`when`.atZone(ZoneId.systemDefault()).toLocalDateTime()
+    return "${whenDateTime.year}/${whenDateTime.monthValue}/${whenDateTime.dayOfMonth}/${auditEvent.who}/" +
```

Comment:
This needs to be formatted as "year=${whenDateTime.year}/month=${whenDateTime.monthValue}/day=${whenDateTime.dayOfMonth}/user=${auditEvent.who}/"